store.go

```diff
@@ -1056,10 +1057,13 @@ func (s *store) imageTopLayerForMapping(image *Image, ristore ROImageStore, crea
 				// We want the layer's parent, too, if it has one.
 				var cParentLayer *Layer
 				if cLayer.Parent != "" {
-					// Its parent should be around here, somewhere.
-					if cParentLayer, err = store.Get(cLayer.Parent); err != nil {
-						// Nope, couldn't find it.  We're not going to be able
-						// to diff this one properly.
+					// Its parent should be in one of the stores, somewhere.
+					for _, ps := range allStores {
+						if cParentLayer, err = ps.Get(cLayer.Parent); err == nil {
```

Comment:
I think the condition should be `err == nil && cParentLayer != nil`. Currently, unless there's an error, we only look at the first store.

Reply:
It will throw an error is the cLayer.Parent does not exists in the store.  This just makes sure that you can find the parent layer if needed.

Reply:
Got it thanks. I did not realize that `err` is declared outside the scope here.